apis/Google.Cloud.Translation.V2/Google.Cloud.Translation.V2/TranslationClientImpl.cs

```diff
@@ -196,12 +203,26 @@ public override IList<Language> ListLanguages(string target = null)
         }
 
         /// <inheritdoc />
-        public override async Task<IList<Detection>> DetectLanguageAsync(string text, CancellationToken cancellationToken = default(CancellationToken))
+        public override async Task<Detection> DetectLanguageAsync(string text, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            GaxPreconditions.CheckNotNull(text, nameof(text));
```

Comment:
I originally had it as a one-liner before adding validation, but by the time you've got:

- The precondition
- ConfigureAwait(false)
- The cancellation token

... it ended up being huge. Happy to combine the two lines around the task though.